File: server/iot.js
const SerialPort = require('serialport');
const Readline = require('@serialport/parser-readline');
const express = require('express');
const app = express();
const http = require('http').Server(app);
const io = require('socket.io')(http);

const portName = '/dev/ttyACM0'; // Change this to match your Arduino's port

const serialPort = new SerialPort(portName, { baudRate: 9600 });
const parser = serialPort.pipe(new Readline({ delimiter: '\n' }));

app.use(express.static('public'));

parser.on('data', (data) => {
  console.log('Data received from Arduino:', data);
  io.emit('dance', data);
});

http.listen(3000, () => {
  console.log('Server is running on port 3000');
});

File: server/iot.py
import serial
import requests

# Configure serial port
ser = serial.Serial('COM5', 9600)  # Replace 'COM3' with the port Arduino is connected to
ser.flushInput()

# Configure server endpoint
url = 'https://pizo-dance.vercel.app/api/data'  # Replace with your server endpoint

try:
    while True:
        if ser.in_waiting > 0:
            # Read data from serial port
            data = ser.readline().decode().strip()
            
            # Extract numerical value from the received data
            voltage_str = data.split(':')[-1].strip().split()[0]
            
            try:
                voltage = float(voltage_str)
                
                # Convert voltage to JSON format
                payload = {'voltage': voltage}
                
                # Send data to server
                response = requests.post(url, json=payload)
                
                if response.status_code == 201:
                    print('Data sent successfully:', voltage)
                else:
                    print('Failed to send data to server:', voltage)
            except ValueError:
                print('Invalid data received from Arduino:', data)
except KeyboardInterrupt:
    ser.close()
    print('Serial port closed')
